py/dml/c_backend.py

```diff
@@ -1977,24 +1964,53 @@ def init_trait_vtables_for_node(node, param_values, dims, parent_indices):
                         if name in method_overrides else "NULL")
                 elif member_kind == 'parameter':
                     # param_overrides contains C expression strings
-                    args.append(param_overrides[name])
+                    (value, t, indexed) = param_overrides[name]
+                    var = f'_param_{name}'
+                    if indexed:
+                        args.append(f'(void *)((uintptr_t){var} + 1)')
+                        array_type = t
+                        for d in reversed(node.dimsizes):
+                            array_type = TArray(array_type,
+                                                mkIntegerLiteral(node.site, d))
+                        param_decl.append(f'''\
+    {TPtr(array_type).declaration(var)} = malloc(sizeof(*{var}));
+    ''')
```

Comment:
I'm not fond of this -- I'd rather concatenate strings together, and at at the proper indentation level. Does The Style have anything to say about this?